core/state_transition.go

```diff
@@ -388,9 +389,9 @@ func (st *StateTransition) TransitionDb() (*ExecutionResult, error) {
 	} else {
 		fee := new(big.Int).SetUint64(st.gasUsed())
 		fee.Mul(fee, effectiveTip)
-		st.state.AddBalance(st.evm.Context.Coinbase, fee)
+		st.state.AddBalance(tipReceipient, fee)
 
-		// Arbitrum: record the tip if nonzero (this should never happen in L2)
+		// Arbitrum: record the tip
 		if st.evm.Config.Debug {
 			st.evm.Config.Tracer.CaptureArbitrumTransfer(st.evm, nil, &st.evm.Context.Coinbase, fee, false, "tip")
```

Comment:
We should probably always record this if `!DropTip()` for clarity (we also record other zero transfers)

Reply:
Moved below so we can do this :)